internal/bgp/bgpfrr/config.go

```diff
@@ -0,0 +1,127 @@
+package bgpfrr
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"text/template"
+)
+
+// TODO configFileName will need to be set to the filename of the FRR configuration
+// file in the container.
+var configFileName = "frr.conf"
+
+const configTemplate = `
+log stdout {{.Loglevel}}
+hostname {{.Hostname}}
+
+{{range .Routers -}}
+router bgp {{.Asn}}
+  no bgp default ipv4-unicast
+  bgp router-id {{.RouterId}}
+{{range .Neighbors }}
+  neighbor {{.Addr}} remote-as {{.Asn}}
+{{- end }}
+{{range $n := .Neighbors -}}
+{{range .Advertisements }}
+  address-family {{.Version}} unicast
+    neighbor {{$n.Addr}} activate
+    network {{.Prefix}}
+  exit-address-family
+{{- end}}
+{{end}}
+{{end}}
+`
+
+type frrConfig struct {
+	Loglevel string
+	Hostname string
+	Routers  map[string]*routerConfig
+}
+
+type routerConfig struct {
+	Asn       uint32
+	RouterId  string
+	Neighbors map[string]*neighborConfig
```

Comment:
I am not sure if this make sense to be map I think slice would have been better here what do u think ?

Reply:
The reason I used maps was that it was a convenient way to remove duplicates. 

For example, two calls to the API New() can create two sessions with the same router but different neighbors that should produce a valid FRR config file something like this: 

```
router bgp 100
  no bgp default ipv4-unicast
  bgp router-id 10.1.1.254

  neighbor 10.2.2.254 remote-as 200
  neighbor 10.4.4.254 remote-as 400
```
Alternatively, perhaps it is possible to structure the FRR config file like:

```
router bgp 100
  no bgp default ipv4-unicast
  bgp router-id 10.1.1.254

  neighbor 10.4.4.254 remote-as 400

router bgp 100
  no bgp default ipv4-unicast
  bgp router-id 10.1.1.254

  neighbor 10.2.2.254 remote-as 200

```
Or, alternatively, I could use a slice to iterate through the routers and look for a duplicate but I think the map will perform better. If you feel strongly about it, I can change it. Alternatively, we can change it in the future if it causes a problem?